needrestart_info.py

```diff
@@ -0,0 +1,164 @@
+            elif key == "NEEDRESTART-KSTA":
+                self.kernel_status = KernelStatus(int(value))
+            # Microcode informations
+            elif key == "NEEDRESTART-UCCUR":
+                self.microcode_current_version = value
+            elif key == "NEEDRESTART-UCEXP":
+                self.microcode_expected_version = value
+            elif key == "NEEDRESTART-UCSTA":
+                self.microcode_status = MicroCodeStatus(int(value))
+            # Count the others
+            else:
+                needrestart_counter.update({key})
+
+        self.services_count = needrestart_counter["NEEDRESTART-SVC"]
+        self.containers_count = needrestart_counter["NEEDRESTART-CONT"]
+        self.sessions_count = needrestart_counter["NEEDRESTART-SESS"]
+
+
+def write_timestamp(registry, needrestart_data):
+    g = Gauge(
+        "needrestart_timestamp",
+        "information about the version and when it was last run",
+        labelnames=["version"],
+        registry=registry,
+    )
+    g.labels(needrestart_data.version).set(needrestart_data.timestamp)
+
+
+def write_kernel(registry, needrestart_data):
+    if needrestart_data.kernel_status:
+        e = Gauge(
+            "needrestart_kernel_status",
+            "information about the kernel status",
+            labelnames=["current", "expected"],
+            registry=registry,
+        )
+        e.labels(
+            needrestart_data.kernel_current_version,
+            needrestart_data.kernel_expected_version,
+        ).set(needrestart_data.kernel_status.value)
+
+
+def write_microcode(registry, needrestart_data):
+    if needrestart_data.microcode_status:
+        e = Gauge(
+            "needrestart_microcode_status",
+            "information about the microcode status",
+            labelnames=["current", "expected"],
+            registry=registry,
+        )
+        e.labels(
+            needrestart_data.microcode_current_version,
+            needrestart_data.microcode_expected_version,
+        ).set(needrestart_data.microcode_status.value)
+
+
+def write_services(registry, needrestart_data):
+    g = Gauge(
+        "needrestart_services_count",
+        "number of services requiring a restart",
+        registry=registry,
+    )
+    g.set(needrestart_data.services_count)
+
+
+def write_containers(registry, needrestart_data):
+    g = Gauge(
+        "needrestart_containers_count",
+        "number of containers requiring a restart",
+        registry=registry,
+    )
+    g.set(needrestart_data.containers_count)
+
+
+def write_sessions(registry, needrestart_data):
+    g = Gauge(
+        "needrestart_sessions_count",
+        "number of sessions requiring a restart",
+        registry=registry,
+    )
+    g.set(needrestart_data.sessions_count)
+
+
+def main():
+    registry = CollectorRegistry()
+
+    needrestart_output = subprocess.run(
+        ["needrestart", "-b"], capture_output=True, text=True
+    ).stdout
```

Comment:
There ought to be some error handling here. What if the command fails to run, i.e. returns non-zero exit status, or what if `subprocess.run()` throws an exception?